parl/remote/remote_decorator.py

```diff
@@ -301,6 +301,14 @@ def __setattr__(self, attr, value):
         return proxy_wrapper
 
     max_memory = kwargs.get('max_memory')
-    if len(args) == 1 and callable(args[0]):
+    """
+        The `args` is different in the following two decorating way, and we should return different wrapper.
```

Comment:
Users may pass some arguments to the decorator (e.g., parl.remote_class(10)). The following code tries to handle this issue.

Reply:
done